app/realtime.php

```diff
@@ -348,13 +263,26 @@ function getCache(): Cache
     while ($attempts < 300) {
         try {
             if ($attempts > 0) {
-                Console::error('Pub/sub connection lost (lasted ' . (time() - $start) . ' seconds, worker: ' . $workerId . ').
-                    Attempting restart in 5 seconds (attempt #' . $attempts . ')');
+                Console::error(
+                    'Pub/sub connection lost (lasted ' . (time() - $start) . ' seconds, worker: ' . $workerId . ').
+                    Attempting restart in 5 seconds (attempt #' . $attempts . ')'
+                );
                 sleep(5); // 5 sec delay between connection attempts
             }
+
             $start = time();
 
-            $redis = $register->get('pools')->get('pubsub')->pop()->getResource(); /** @var Redis $redis */
+            $pools = $container->get('pools');
+            /** @var Connections $connections */
+            $connections = $container->get('connections');
+
+            $pool = $pools['pools-pubsub-pubsub']['pool'];
+            $connection = $pool->get();
+            $connections->add($connection, $pool);
+
+            $redis = $connection;
```

Comment:
```suggestion
            $pools = $container->get('pools');
            $pool = $pools['pools-pubsub-pubsub']['pool'];
            
            /** @var Connections $connections */
            $connections = $container->get('connections');
            $connection = $pool->get();
            $connections->add($connection, $pool);

            $redis = $connection;
```
I would just rewrite this differently